### PR TITLE
Create avatar composable

### PR DESCRIPTION
**Pull request by RomainGF**

### CODE REVIEW

File: library-compose/src/main/java/com/spendesk/grapes/compose/avatar/Avatar.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        model = avatar,
+        contentDescription = contentDescription,
+        placeholder = placeholder,
+        alignment = alignment,
+        contentScale = contentScale,
+        alpha = alpha,
+        colorFilter = colorFilter,
+        modifier = modifier
+            .clip(CircleShape)
+            .clickable(enabled = onClick != null, onClick = { onClick?.invoke() }),
```
Comment (jpbarbaud) [inline library-compose/src/main/java/com/spendesk/grapes/compose/avatar/Avatar.kt:44]: Grapes should be agnostic from external lib and should not be responsible to download the avatar itself. 
We should use a wrapper/imageProvider or something else to download the image. 
What do you think?

Comment (RomainGF) [reply library-compose/src/main/java/com/spendesk/grapes/compose/avatar/Avatar.kt:44]: Or maybe I can let a slot on the EditAvatar composable to keep the image loading on Spendesk app. What do you think about that @jpbarbaud ?

Comment (kelianClerc) [reply library-compose/src/main/java/com/spendesk/grapes/compose/avatar/Avatar.kt:44]: We could add a slot that takes a modifier as parameter. The modifier being the one you used here.
But by doing that, can we specify the size of the avatar like you did in the preview ?

File: library-compose/src/main/java/com/spendesk/grapes/compose/avatar/EditAvatar.kt [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+private const val LAYOUT_ID_AVATAR_EDIT = "avatar-edit"
+
+@Composable
+fun EditAvatarLayout(
+    badge: @Composable BoxScope.() -> Unit,
+    avatar: @Composable BoxScope.() -> Unit,
+    modifier: Modifier = Modifier,
+    badgeOffset: DpOffset,
+) {
+    Layout(
```
Comment (kelianClerc) [inline library-compose/src/main/java/com/spendesk/grapes/compose/avatar/EditAvatar.kt:85]: Beautiful!

File: library-compose/src/main/java/com/spendesk/grapes/compose/avatar/EditAvatar.kt [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                this.contentDescription = contentDescription
+            },
+    )
+}
+
+private const val LAYOUT_ID_AVATAR = "avatar-anchor"
+private const val LAYOUT_ID_AVATAR_EDIT = "avatar-edit"
+
+@Composable
+fun EditAvatarLayout(
```
Comment (kelianClerc) [inline library-compose/src/main/java/com/spendesk/grapes/compose/avatar/EditAvatar.kt:79]: Shouldn't this composable be private? Or at least internal.
The question underlying is: do we want Spendesk to be able to use that layout to show an Avatar in a way Grapes did not planned. 
Imo the answer is no. But I'm open to a debate

Comment (RomainGF) [reply library-compose/src/main/java/com/spendesk/grapes/compose/avatar/EditAvatar.kt:79]: My bad I agree this composable must be private

File: library-compose/src/main/java/com/spendesk/grapes/compose/avatar/Avatar.kt (hunk 1 above)
Comment (RomainGF) [reply library-compose/src/main/java/com/spendesk/grapes/compose/avatar/Avatar.kt]: Yes we can use fillMaxSize() on the avatar slot modifier and `width(IntrinsicSize.Min) & height(IntrinsicSize.Min)` on the main modifier. Here's the result with a square blue box in the slot
Capture d’écran 2023-06-21 à 10 55 09

**Review by kelianClerc — APPROVED** [review]

File: library-compose/src/main/java/com/spendesk/grapes/compose/avatar/GrapesEditAvatar.kt [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+            val badgeY = avatarRadius * sin(angle) - editRadius + avatarRadius
+            editPlaceable.placeRelative(
+                x = badgeX.toInt() + badgeOffset.x.toPx().toInt(),
+                y = badgeY.toInt() + badgeOffset.y.toPx().toInt(),
+            )
+        }
+    }
+}
+
+object EditAvatarDefaults {
```
Comment (jpbarbaud) [inline library-compose/src/main/java/com/spendesk/grapes/compose/avatar/GrapesEditAvatar.kt:121]: add `@Immutable` annotation

**Review by jpbarbaud — APPROVED** [review]

**Pull request merged**